downloader_class.py

```diff
@@ -0,0 +1,131 @@
+'''
+Author: Keenan Manpearl
+Date Created: 1/30/2023
+'''
+
+
+import pandas as pd
+import pathlib
+import os
+
+
+class AsperaDownloader:
+    """
+    This class holds all functions needed to download IDR image data with Aspera
+
+    Attributes
+    ----------
+    aspera_path : pathlib.Path
+        path to aspera sdk
+    aspera_key_path : pathlib.Path
+        path to aspera openssh key
+    screens : pd.Dataframe
+        dataframe with curated IDR screen data for specific study
+    idr_id: str
+        accession ID for IDR study
+
+    Methods
+    -------
+    get_IDR_mitocheck_movie_path(plate, well_num)
+        get image path for Aspera download by finding screen for specific image in screen metadata
+    download_image(plate, well_num)
+        download image corresponding to plate and well number with Aspera
+
+    Example
+    -------
+    import pathlib
+    from IDR_stream.download import AsperaDownloader
+
+    aspera_path = pathlib.Path("/home/user/.aspera/ascli/sdk/ascp")
+    aspera_key_path = pathlib.Path("asperaweb_id_dsa.openssh")
+    screens_path = pathlib.Path("idr0013-screenA-plates.tsv")
+    idr_id = "idr0013"
+    save_dir = pathlib.Path("downloaded_images/")
+
+    downloader = AsperaDownloader(aspera_path, aspera_key_path, screens_path, idr_id)
+    downloader.download_image("LT0001_02", 4, save_dir)
+    """
+
+    def __init__(
+        self,
+        aspera_path: pathlib.Path,
+        aspera_key_path: pathlib.Path,
+        screens_path: pathlib.Path,
+        idr_id: str,
+    ):
+        """
+        __init__ function for AsperaDownloader class
```

Comment:
Functions defined within the context of a class are called "methods" (`__init__` being a special constructor method for initializing the class). This is a minor detail, but can provide useful detail when reading/speaking about specific aspects of your code.

```suggestion
        __init__ method for AsperaDownloader class
```